File: engine/model/src/main/scala/tools/packet/alliance/RemoveGuildFromAlliance.scala
package tools.packet.alliance

import net.opcodes.SendOpcode
import net.server.guild.MapleAlliance
import tools.packet.PacketInput

class RemoveGuildFromAlliance(private var _alliance: MapleAlliance, private var _expelledGuildId: Int, private var _worldId: Int) extends PacketInput {
  def alliance: MapleAlliance = _alliance

  def expelledGuildId: Int = _expelledGuildId

  def worldId: Int = _worldId

  override def opcode(): SendOpcode = SendOpcode.ALLIANCE_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/alliance/RemoveGuildFromAlliance.java
package tools.packet.alliance;

import net.opcodes.SendOpcode;
import net.server.guild.MapleAlliance;
import tools.packet.PacketInput;

public record RemoveGuildFromAlliance(MapleAlliance alliance, Integer expelledGuildId,
                                      Integer worldId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ALLIANCE_OPERATION;
   }
}
